### PR TITLE
base changes for issue #338

### PR DESCRIPTION
**Pull request by lukmdo**
For issue https://github.com/zaquestion/lab/issues/338

## main changes 
- use of `sha` instead of `ref/branchName` https://docs.gitlab.com/ee/api/pipelines.html#list-project-pipelines is more correct with detached pipelines ( via `only: merge_requests` )
- use `sha` of upstream (not local)

## additionally  
- add `GetBranch` and remove/replace `BranchPushed`  
- `cmd` parse args unification
  - use `parseArgsRemoteString`
  - add `parseArgsRemoteRef` (helper)

### CODE REVIEW

File: cmd/ci_view.go [hunk 1]
```diff
@@ -77,6 +78,11 @@ Feedback Encouraged!: https://github.com/zaquestion/lab/issues`,
 			log.Fatal(err)
 		}
 		projectID = project.ID
+		branch, err := lab.GetBranch(projectID, branchName)
```
Comment (lukmdo) [inline cmd/ci_view.go:81]: this needs redo as before `view [remote [branch/tag]]` was accepting both as ref not only  branchName

I am thinking of sth like 
```golang
func FindCommitByRef(pid interface{}, ref string) (*gitlab.Commit, error) {}
```
via `lab.Commits.ListCommits` https://docs.gitlab.com/ce/api/commits.html#list-commits

Comment (lukmdo) [reply cmd/ci_view.go]: done on https://github.com/zaquestion/lab/pull/380/commits/468bb6916394a58f32ee962b98cdbdbbefac920f

**Comment by lukmdo** [comment]
@zaquestion help with review - hope you will like this

File: cmd/root_test.go [hunk 2]
```diff
@@ -404,6 +404,73 @@ func Test_parseArgsRemoteString(t *testing.T) {
 	}
 }
 
+func Test_parseArgsRemoteRef(t *testing.T) {
```
Comment (lukmdo) [inline cmd/root_test.go:407]: not too DRY (based on Test_parseArgsRemoteString) - but should be still OK ?

**Comment by zaquestion** [comment]
@lukmdo this looks good, I originally implemented the ci jobs stuff as sha and switched away since I ran into issues retrieving the latest pipelines on reruns or something like that. I think the issues I hit were things like generally not seeing the jobs/pipelines you'd expect if you were viewing the CI on the MR or in the UI.

That said, your implementation looks stronger than what I originally crapped out, so I'm inclined to bring this in and see how it goes.

**Comment by zaquestion** [comment]
Oh now I see, and remember that we already discussed a lot of this in https://github.com/zaquestion/lab/issues/338

**Pull request merged**